src/crawlee/storage_clients/_sql/_request_queue_client.py

```diff
@@ -0,0 +1,619 @@
+
+        raise NotImplementedError('Batch addition is not supported for this database dialect.')
+
+    @override
+    async def get_request(self, unique_key: str) -> Request | None:
+        if not (request_id := self._REQUEST_ID_BY_KEY.get(unique_key)):
+            request_id = self._get_int_id_from_unique_key(unique_key)
+            self._REQUEST_ID_BY_KEY[unique_key] = request_id
+
+        stmt = select(self._ITEM_TABLE).where(
+            self._ITEM_TABLE.metadata_id == self._id, self._ITEM_TABLE.request_id == request_id
+        )
+        async with self.get_session() as session:
+            result = await session.execute(stmt)
+            request_db = result.scalar_one_or_none()
+
+            if request_db is None:
+                logger.warning(f'Request with ID "{unique_key}" not found in the queue.')
+                return None
+
+            updated = await self._update_metadata(session, update_accessed_at=True)
+
+            # Commit updates to the metadata
+            if updated:
+                await session.commit()
+
+        return Request.model_validate_json(request_db.data)
+
+    @override
+    async def fetch_next_request(self) -> Request | None:
+        if self._pending_fetch_cache:
+            return self._pending_fetch_cache.popleft()
+
+        now = datetime.now(timezone.utc)
+        block_until = now + timedelta(seconds=self._BLOCK_REQUEST_TIME)
+        dialect = self._storage_client.get_dialect_name()
+
+        # Get available requests not blocked by another client
+        stmt = (
+            select(self._ITEM_TABLE)
+            .where(
+                self._ITEM_TABLE.metadata_id == self._id,
+                self._ITEM_TABLE.is_handled.is_(False),
+                or_(self._ITEM_TABLE.time_blocked_until.is_(None), self._ITEM_TABLE.time_blocked_until < now),
+            )
+            .order_by(self._ITEM_TABLE.sequence_number.asc())
+            .limit(self._MAX_BATCH_FETCH_SIZE)
+        )
+
+        # We use the `skip_locked` database mechanism to prevent the “interception” of requests by another client
+        if dialect == 'postgresql':
+            stmt = stmt.with_for_update(skip_locked=True)
+
+        async with self.get_session() as session:
+            result = await session.execute(stmt)
+            requests_db = result.scalars().all()
+            if not requests_db:
+                return None
+
+            request_ids = {r.request_id for r in requests_db}
+
+            # Mark the requests as blocked
+            update_stmt = (
+                update(self._ITEM_TABLE)
+                .where(
+                    self._ITEM_TABLE.metadata_id == self._id,
+                    self._ITEM_TABLE.request_id.in_(request_ids),
+                    self._ITEM_TABLE.is_handled.is_(False),
+                    or_(self._ITEM_TABLE.time_blocked_until.is_(None), self._ITEM_TABLE.time_blocked_until < now),
+                )
+                .values(time_blocked_until=block_until)
```

Comment:
It would be helpful to use something like the `client_key` in the Apify request queue client here so that we know _who_ locked the request. That way, the storage client would be more resilient against migrations.